### DISCUSSION TITLE
Styling module.css using JS

### DISCUSSION THREAD

**Discussion by Dave-Patsy** [vercel/next.js#40382] [Help] [answered]
I was hoping that I could interact with Style.circle and change the properties of it using javascript. However, it seems that is not the case. would I just have to use JSX style to change the position that way?

```
import { useEffect } from "react"
import Style from '../../../styles/games/circle/circle.module.css'

const Circle = () => {

    useEffect(()=>{
        const handleClick = event => {
            Style.circle.transform = `translateY(${event.clientY - 25}px)`
            Style.circle.transform += `translateX(${event.clientX - 25}px)`
        }

        window.addEventListener('click',handleClick,false)

        return () => {
            window.removeEventListener('click',handleClick,false)
        }
    },[])

    return(
        <div className={Style.circle}>

        </div>
    )
}

export default Circle
```

this is the circle.module.css
```
.circle{
    position: absolute;
    background: red;
    border-radius: 25px;
    width: 50px;
    height: 50px;
    /* top: 50%;
    left: 50%; */
    transform: none;
    transition: transform 1s;
}
```
![Nextjs click](https://user-images.githubusercontent.com/24628331/189274180-799bb358-5b06-41a9-9469-7b55150bf322.PNG)

Lastly, I tried
```
    useEffect(()=>{

    const f = document.getElementById(Style.circle);
    console.log('this is the className of: ',f)
        const handleClick = event => {
            f.style.transform = `translateY(${event.clientY - 25}px)`
            f.style.transform += `translateX(${event.clientX - 25}px)`
        }
    ....
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The pitfall here is that we may perceive CSS Modules, as objects, but they are just classes (strings). The CSS syntax through the imports statements might result in confusion. CSS modules are, strictly speaking, not a CSS in JS kind of solution.

`Style.circle` is a string that represents a CSS class, and it tells the browser which styles to use from stylesheets or style tags.

In this case you can either, add a ref to the `div`, and change its style imperatively, or you could use `useState` to save the new transform value, trigger a re-render and via `CSS vars` pass the new transform as `style` prop. I'd go for the `ref` and avoid having to manage state:

```jsx
const Circle = () => {
  const ref = useRef(null);

  useEffect(() => {
    const handleClick = (event) => {
      const element = ref.current;
      if (!element) return;

      const transform = `translateY(${event.clientY - 25}px) translateX(${event.clientX - 25}px)`;
      element.style.transform = transform;
    };

    window.addEventListener("click", handleClick, false);

    return () => {
      window.removeEventListener("click", handleClick, false);
    };
  }, []);

  return <div className={Style.circle} ref={ref}></div>;
};

```

**Reply by Dave-Patsy** [reply]
Thank You icyJoseph. I haven't really used the ref hook except in react-three-fiber. This helps a lot!
